packages/frameworks/vue/src/dialog/use-dialog.ts

```diff
@@ -1,28 +1,33 @@
 import * as dialog from '@zag-js/dialog'
 import { normalizeProps, useMachine, type PropTypes } from '@zag-js/vue'
-import { computed, reactive, watch, type ComputedRef } from 'vue'
+import { computed, ref, type ComputedRef } from 'vue'
 import { useEnvironmentContext } from '../environment'
 import type { Optional } from '../types'
 import { useId } from '../utils'
 
 export interface UseDialogProps extends Optional<dialog.Context, 'id'> {
-  modelValue?: dialog.Context['open']
+  /**
+   * The initial open state of the dialog.
+   */
+  defaultOpen?: dialog.Context['open']
+  'onUpdate:open'?: (open: dialog.OpenChangeDetails['open']) => void
 }
 
 export interface UseDialogReturn extends ComputedRef<dialog.Api<PropTypes>> {}
 
 export const useDialog = (props: UseDialogProps, emit: CallableFunction) => {
+  const context = ref(props)
   const getRootNode = useEnvironmentContext()
-  const reactiveContext = reactive(props)
 
   const [state, send] = useMachine(
     dialog.machine({
-      ...reactiveContext,
-      id: reactiveContext.id || useId().value,
+      ...context.value,
+      id: context.value.id || useId().value,
+      open: props.open ?? props.defaultOpen,
```

Comment:
I found this in the Checkbox, should we change it to `checked: props.checked ?? props.defaultChecked` too?

![image](https://github.com/chakra-ui/ark/assets/6079265/9d142c48-273c-4c61-a5de-6b2979023807)

Reply:
yep, i missed it

Reply:
I'll fix that